lib/overlap/holding_commitment.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "cluster"
+require "services"
+
+Services.mongo!
+
+module Overlap
+  # Given an ocn, gets:
+  # * commitments in the matching cluster
+  # * holdings eligible for commitments in the matching cluster
+  # * all matched up pairs of holdings-commitments from the above
+  class HoldingCommitment
+    attr_reader :holdings, :commitments
+
+    def initialize(ocn)
+      @holdings    = []
+      @commitments = []
+
+      # For holding-commitment overlaps we only care about spm clusters with ht_items
+      Cluster.where(ocns: ocn, "ht_items.0": { "$exists": 1 }).each do |cluster|
+        if cluster.format == "spm"
+          @holdings    = filter_holdings(cluster.holdings)
+          @commitments = filter_commitments(cluster.commitments)
+        end
+      end
+    end
+
+    # How many distinct orgs have reported holdings for the given cluster?
+    def holdings_h
+      @holdings.pluck(:organization).uniq.count
+    end
+
+    # How many distinct orgs have reported commitments for the given cluster?
+    def commitments_h
+      @commitments.pluck(:organization).uniq.count
+    end
+
+    # Return only holdings that could have a commitment.
+    def filter_holdings(holdings)
+      holdings.select {|x| x.mono_multi_serial == "mono" && (x.status.empty? || x.status == "CH") }
```

Comment:
Might be worth some more explanation in comments here -- if the member doesn't report it as `mono_multi_serial` then something has gone wrong with their data; `CH` status is current holding & we don't want a commitment on something lost/missing/withdrawn.